### PR TITLE
Fixes #30737 - Add VLAN tag to Ubuntu preseed if present

### PR DESCRIPTION
**Pull request by stefanlasiewski**
Add code to pass parameters from the Kernel commandline to preseed/debian-installer to add a VLAN if the VLAN tag is present on a network.

~I have an ERB question and will ask it in the code review.~ Resolved.

### CODE REVIEW

File: app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb [hunk 1]
```diff
@@ -28,6 +28,10 @@ oses:
   else
     options << 'console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true'
   end
+  if @host.provision_interface.vlanid
+    vlan_opt = "netcfg/use_vlan=true netcfg/vlan_id=#{@host.provision_interface.vlanid}"
+    options << "#{vlan_opt}"
+  end
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb:33]: Can you make this just one line? Also squash everything into a single commit if possible. I haven't tested but it looks good.

Comment (stefanlasiewski) [reply app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb:33]: Done. I assume you still want the `if` / `end` syntax so it's consistent with the other conditionals in this block.

**Comment by ekohl** [comment]
Would you mind shortening the commit title a bit? https://chris.beams.io/posts/git-commit/ has more info, but you can use paragraphs.

**Comment by stefanlasiewski** [comment]
> Would you mind shortening the commit title a bit?

Done.

**Comment by tbrisker** [comment]
@stefanlasiewski looks like there was some issue with your rebase, and it pulled in a bunch of other commits. could you `git rebase develop; git push -f` on your branch please to get it sorted?

**Comment by ekohl** [comment]
Besides what @tbrisker mentioned, this looks good.

**Comment by stefanlasiewski** [comment]
> @stefanlasiewski looks like there was some issue with your rebase, and it pulled in a bunch of other commits. could you `git rebase develop; git push -f` on your branch please to get it sorted?

Oh no, I don't know what I did. I need to dig myself out of this a bit.  It needs more than `git rebase develop; git push -f`. 

I might not be able to get back to this until later this week.

(Sorry, I don't do this with git very often.)

**Comment by tbrisker** [comment]
No worries @stefanlasiewski, I fixed it for you :) you'll need to use `git pull -f` though to get the updated version into your local branch if needed.

**Comment by tbrisker** [comment]
ok to test

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/PXELinux/Preseed default PXELinux.snap.txt [hunk 2]
```diff
@@ -15,7 +15,7 @@ DEFAULT linux
 
 LABEL linux
     KERNEL 
-    APPEND initrd= interface=auto url=http://foreman.some.host.fqdn/unattended/provision ramdisk_size=10800 root=/dev/rd/0 rw auto hostname=snapshothost console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true locale=en_US
+    APPEND initrd= interface=auto url=http://foreman.some.host.fqdn/unattended/provision ramdisk_size=10800 root=/dev/rd/0 rw auto hostname=snapshothost console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true netcfg/use_vlan=true netcfg/vlan_id= locale=en_US
```
Comment (tbrisker) [inline test/unit/foreman/renderer/snapshots/ProvisioningTemplate/PXELinux/Preseed default PXELinux.snap.txt]: @stefanlasiewski I've updated the snapshot for you but this doesn't seem correct?

File: app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb [hunk 3]
```diff
@@ -28,6 +28,9 @@ oses:
   else
     options << 'console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true'
   end
+  if @host.provision_interface.vlanid
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb:31]: Looks like there is a `has_vlanid?` helper that's also allowed in the jail
```suggestion
  if @host.provision_interface.has_vlanid?
```

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/PXELinux/Preseed default PXELinux.snap.txt (hunk 2 above)
Comment (lzap) [reply test/unit/foreman/renderer/snapshots/ProvisioningTemplate/PXELinux/Preseed default PXELinux.snap.txt]: This is not correct, I just merged @ekohl 's change which fixes this. Let me fix this.

**Comment by lzap** [comment]
I squashed and fixed tests in https://github.com/theforeman/foreman/pull/8201

**Comment by lzap** [comment]
The change was already merged for Foreman 2.4 in https://github.com/theforeman/foreman/pull/8201/files

I am closing this PR, thank you!

**Pull request closed without merging**